astropy/units/physical.py

```diff
@@ -1,133 +1,425 @@
-    if r in _physical_unit_mapping:
+
+    @staticmethod
+    def _standardize_physical_type_names(
+        physical_type_input: Union[str, Set[str]]
+    ) -> Set[str]:
+        """
+        Convert a string or `set` of strings into a set containing string
+        representations of physical types.
+
+        The strings provided in ``physical_type_input`` can each contain
+        multiple physical types that are separated by a backslash or
+        comma.  Underscores will be treated as spaces.
+        """
+
+        if not isinstance(physical_type_input, (set, str)):
+            raise TypeError(
+                f"Expecting a string or a set containing strings that "
+                f"represent a physical type, not {physical_type_input}"
+            )
+
+        if isinstance(physical_type_input, str):
+            physical_type_input = {physical_type_input}
+
+        delimiter_pattern = ",|/"
+        standardized_physical_types = set()
+
+        for ptype_input in physical_type_input:
+            if not isinstance(ptype_input, str):
+                raise TypeError(f"Expecting a string, but got {ptype_input}")
+            input_set = set(re.split(delimiter_pattern, ptype_input))
+            processed_set = {s.strip().replace("_", " ") for s in input_set}
+            standardized_physical_types |= processed_set
+
+        return standardized_physical_types
+
+    def __init__(self, unit, physical_types: Union[str, Set[str]]):
+
+        if not isinstance(unit, core.UnitBase):
+            raise TypeError(f"Expecting a unit, not {unit}.")
+
+        self._unit = _replace_temperatures_with_kelvin(unit)
+        self._physical_type_id = self._unit._get_physical_type_id()
+        self._as_set = self._standardize_physical_type_names(physical_types)
+
+    @property
+    def as_set(self) -> Set[str]:
+        """Return a `set` of all physical types that represent a `Unit`."""
+        return self._as_set
+
+    @property
+    def as_string(self) -> str:
+        """Return a string representation of the physical type(s)."""
+        if len(self.as_set) == 1:
+            return list(self.as_set)[0]
+        else:
+            return "{" + str(sorted(self.as_set))[1:-1] + "}"
+
+    def _get_physical_type_id(self) -> Tuple[Tuple[str, numbers.Integral]]:
+        """
+        Returns an identifier that uniquely identifies the physical type
+        of this unit.  See `core.Unit._get_physical_type_id`.
+        """
+        # To match Unit's internal API
+        return self._physical_type_id
+
+    def __eq__(self, other):
+        """
+        Return `True` if ``other`` represents a physical type that is
+        consistent with the physical type of the `PhysicalType` instance.
+        """
+        if isinstance(other, (str, set)):
+            other = self._standardize_physical_type_names(other)
+
+        if isinstance(other, set):
+            return other.issubset(self.as_set)
+        elif isinstance(other, PhysicalType):
```

Comment:
One of those totally trivial things, but I'd put this first as it is the comparison that should be easiest to deal with.